### PR TITLE
Adding feature to handle batches that results in error on copy

### PR DESCRIPTION
**Pull request by trisberg**
- adding transaction management

- retrying failed batch by copying each row individually

- logging errors

- add property for error-table

- writing errors to table if error-table specified

### CODE REVIEW

File: spring-cloud-starter-stream-sink-pgcopy/src/main/java/org/springframework/cloud/stream/app/pgcopy/sink/PgcopySinkConfiguration.java [hunk 1]
```diff
@@ -145,32 +153,88 @@ public void handleMessage(Message<?> message) throws MessagingException {
...
+						logger.error("Switching to single row copy for current batch");
+						long rows = 0;
+						for (Object singlePayload : payloads) {
+							try {
+								rows = rows + doCopy(Collections.singletonList(singlePayload), txTemplate);
+							}
+							catch (DataAccessException e2) {
+								logger.error("Copy for single row caused error: " + e2.getMessage());
+								logger.error("Bad Data: \n" + singlePayload);
+								System.err.println("Bad Data: " + singlePayload);
```
Comment (jvalkeal) [inline spring-cloud-starter-stream-sink-pgcopy/src/main/java/org/springframework/cloud/stream/app/pgcopy/sink/PgcopySinkConfiguration.java]: what's a reasoning writing to stderr?

Comment (trisberg) [reply spring-cloud-starter-stream-sink-pgcopy/src/main/java/org/springframework/cloud/stream/app/pgcopy/sink/PgcopySinkConfiguration.java]: just having another source for the bad rows

**Comment by jvalkeal** [comment]
well works as advertised, merging.

**Comment by jvalkeal** [comment]
Squashed/merged per https://github.com/spring-cloud-stream-app-starters/jdbc/commit/2c26667f1d82f50143c6fdf26f86a4fde33ed349

**Pull request closed without merging**